apps/sms/js/thread_ui.js

```diff
@@ -598,10 +598,11 @@ var ThreadUI = {
       });
     }
 
-    ThreadListUI.mark(threadId, 'read');
-
-    // nothing urgent, let's do it when the main thread has some time
-    setTimeout(MessageManager.markThreadRead.bind(MessageManager, threadId));
```

Comment:
I think we still want a setTimeout so that the browser can do its reflows as soon as possible; (remember: promises are microtasks, not macrotasks). I'm not sure this has a real effect but this could on a busy phone.